test/generators/shop_model_generator_test.rb

```diff
@@ -29,6 +33,13 @@ class ShopModelGeneratorTest < Rails::Generators::TestCase
     end
   end
 
+  test "injects into application controller" do
+    run_generator
+    assert_file "app/controllers/application_controller.rb" do |controller|
+      assert_match "def shop", controller
+    end
```

Comment:
Indeed. I just added the shop method and all works as advertised. Sorry for the intrusion on the old thread. Just wanted to clear the air a bit on this issue. It's critical for my Apps to have *shop* at my disposal throughout the Apps I build. I can understand though why there can be a circus around this for implementation.

*Thanks* for the App!